### ISSUE TITLE
Saplo Client initiates session in constructor

### ISSUE THREAD

**Issue by chids** [saplo/saplo4j#1]
The call to [createSession in SaploClient](https://github.com/saplo/saplo4j/blob/master/src/main/java/com/saplo/api/client/SaploClient.java#L145) prohibits SaploClient from being constructed with an endpoint that is not available at instantiation time.

This makes testing awkward since third party tests might require a SaploClient instance but won't actually invoke the Saplo API. For such tests we wan't to give a bogus URL such as _http://non.existant_ to make sure we don't invoke any Saplo APIs.

**Comment by horte** [comment]
Added a builder option 'connect' which if set to false will not call createSession.

See https://github.com/saplo/saplo4j/commit/572f01b58066cabd5a34de9fd95f81fb9d2fe12d 
Would that solve your problem? 

```
client = new SaploClient.Builder("abc123","abc123").endpoint("http://foobar.com").connect(false).build();
```

**Comment by chids** [comment]
:raised_hands:

**Comment by horte** [comment]
I'll take that as a yes and will push the changes to maven.

**Comment by chids** [comment]
:sparkling_heart:

**Comment by chids** [comment]
FYI: @kristiany, @peterlind, @hnrc

**Comment by chids** [comment]
and @andershedstrom...

**Comment by horte** [comment]
Released to maven as version 1.0.7. Waiting for repo to sync.

**Comment by chids** [comment]
:+1:
